mp4parse_capi/src/lib.rs

```diff
@@ -853,6 +837,37 @@ impl<'a> Iterator for SampleToChunkIterator<'a> {
     }
 }
 
+impl<'a> SampleToChunkIterator<'a> {
+    fn locate(&mut self) -> std::ops::Range<u32> {
+        loop {
+            let r = match (self.stsc_peek_iter.next(), self.stsc_peek_iter.peek()) {
```

Comment:
This might be a bit cleaner as:

```
+        loop {
+            return match (self.stsc_peek_iter.next(), self.stsc_peek_iter.peek()) {
+                (Some(next), Some(peek)) if next.first_chunk == peek.first_chunk =>
+                    // Invalid entry, skip it and will continue searching at
+                    // next loop iteration.
+                    continue,
+                (Some(next), Some(peek)) if next.first_chunk > 0 && peek.first_chunk > 0 => {
+                    self.sample_count = next.samples_per_chunk;
+                    (next.first_chunk - 1) .. (peek.first_chunk - 1)
+                },
+                (Some(next), None) if next.first_chunk > 0 => {
+                    self.sample_count = next.samples_per_chunk;
+                    // Total chunk number in 'stsc' could be different to 'stco',
+                    // there could be more chunks at the last 'stsc' record.
+                    match next.first_chunk.checked_add(self.remain_chunk_count) {
+                        Some(r) => (next.first_chunk - 1) .. r - 1,
+                        _ => 0 .. 0,
+                    }
+                },
+                _ => 0 .. 0
+            };
+        };
```

And please add a test for the new case.

Reply:
same as above...
can add the init segment only and parse that to see if we have any samples...